hkmc2/shared/src/main/scala/hkmc2/codegen/BlockTraverser.scala

```diff
@@ -68,6 +68,8 @@ class BlockTraverser(subst: SymbolSubst):
     case Value.Lit(lit) => ()
     case v @ Value.Lam(params, body) => applyLam(v)
     case Value.Arr(elems) => elems.map(applyArg)
+    case Value.Rcd(fields) => fields.map:
```

Comment:
Wait why is the `BlockTraverser` using `map` everywhere? That's crazy 😅 

It should use `foreach` to avoid building useless intermediate data structures!